src/preprocess/dataset_split/data_subset.py

```diff
@@ -0,0 +1,42 @@
+import os
+import pandas as pd
+from sklearn.model_selection import GroupShuffleSplit
```

Comment:
I think we should use the StratifiedShuffleSplit here. GroupShuffleSplit does not guarantee that the class distribution is preserved in the splits